examples/test-exec-doc.yml

```diff
@@ -0,0 +1,221 @@
+      id: check_changes
+      run: |
+        # Check if this is the initial commit (all zeros SHA)
+        if [ "${{ github.event.before }}" = "0000000000000000000000000000000000000000" ]; then
+          echo "Initial commit detected."
+
+          # Determine the default branch (e.g., main)
+          if git show-ref --verify --quiet refs/remotes/origin/main; then
+            DEFAULT_BRANCH="main"
+          elif git show-ref --verify --quiet refs/remotes/origin/master; then
+            DEFAULT_BRANCH="master"
+          else
+            echo "Default branch is not 'main' or 'master'. Getting all Markdown files."
+            CHANGED_MARKDOWN_FILES=$(git ls-files '*.md')
+            echo "Markdown files to test:"
+            echo "$CHANGED_MARKDOWN_FILES"
+            echo "changed=true" >> $GITHUB_OUTPUT
+            echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+            echo "$CHANGED_MARKDOWN_FILES" >> $GITHUB_OUTPUT
+            echo "EOF" >> $GITHUB_OUTPUT
+            exit 0
+          fi
+          echo "Default branch is $DEFAULT_BRANCH"
+          
+          # Get the last commit SHA of the default branch
+          UPSTREAM_SHA=$(git rev-parse origin/$DEFAULT_BRANCH)
+          echo "Upstream SHA: $UPSTREAM_SHA"
+          
+          # List changed Markdown files by comparing with the upstream branch
+          CHANGED_MARKDOWN_FILES=$(git diff --name-only $UPSTREAM_SHA ${{ github.sha }} | grep -E '\.md$' || true)
+
+        else
+          # Compare the latest commit with the previous one
+          CHANGED_FILES=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }})
+          echo "Changed files: $CHANGED_FILES"
+          # Filter changed Markdown files
+          CHANGED_MARKDOWN_FILES=$(echo "$CHANGED_FILES" | grep -E '\.md$' || true)
+        fi
+
+        echo "Markdown files to test:"
+        echo "$CHANGED_MARKDOWN_FILES"
+
+        if [ ! -z "$CHANGED_MARKDOWN_FILES" ]; then
+          echo "changed=true" >> $GITHUB_OUTPUT
+          # Set multiline output using the '<<EOF' syntax
+          echo "changed_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$CHANGED_MARKDOWN_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+        else
+          echo "changed=false" >> $GITHUB_OUTPUT
+          echo "No Markdown files changed. Exiting."
+          exit 0
+        fi
+        
+    - name: Azure CLI Login
+      if: steps.check_changes.outputs.changed == 'true'
+      uses: azure/login@v2
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}             # Azure Client ID from secrets
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}             # Azure Tenant ID from secrets
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }} # Azure Subscription ID from secrets
+
+    - name: Set up Python
+      if: steps.check_changes.outputs.changed == 'true'
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'  # Specify Python version 3.x
```

Comment:
Please pin this to python 3.12, there's no need to have this pipeline always using the latest python

Reply:
done - will add fallback here as well for posterity